commands/global/freq.c

```diff
@@ -333,10 +333,8 @@ const uint max_gate_time_ms = 100;
 static float freq_measure_reciprocal_and_duty_cycle(uint gpio, float* duty)
 {
     PIO pio = pio0;
-    uint sm = 0;
+    int sm = pio_claim_unused_sm(pio, true);

```

Comment:
if you change to the `int` to reflect the usage in the API it looks like it's still not checked later on when releasing. it's more of a nit than anything since the rest of the code doesn't seem to check to make sure the state machine was acquired either, but it makes me wonder what `pio_sm_unclaim` does if a -1 is passed in for whatever reason since only 0-3 are valid.

Reply:
Ah... This will panic instead of returning -1 because the required flag is passed in. @DangerousPrototypes raised the question of limited resources (like PIO, SM, PIO programs) and what to do about it. I personally prefer to panic rather than re-use a SM that belongs to another piece of program currently running.

Reply:
I don't consider the discussion closed. This is more of a first step on the read of resource management

Reply:
That makes sense; it was really more of a curiosity for me since I'm still working to understand how the PIOs